### PR TITLE
Fixing docker fake client - InspectExec

### PR DESCRIPTION
**Pull request by krzysied**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Prevents docker nil pointer panics when fake client is used.

**Which issue(s) this PR fixes**:

ref #84855

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/kubelet/dockershim/libdocker/fake_client.go [hunk 1]
```diff
@@ -762,7 +762,10 @@ func (f *FakeDockerClient) AttachToContainer(id string, opts dockertypes.Contain
 }
 
 func (f *FakeDockerClient) InspectExec(id string) (*dockertypes.ContainerExecInspect, error) {
-	return f.ExecInspect, f.popError("inspect_exec")
+	if f.ExecInspect != nil {
```
Comment (mattjmcnaughton) [inline pkg/kubelet/dockershim/libdocker/fake_client.go]: From looking at the code, I'm confused about if the `f.ExecInspect != nil` boolean will ever evaluate to `True`. Looking in `pkg/kubelet/dockershim` I'm not seeing any instances where that value is actually set.

Should we either assign `f.ExecInspect` to be `&dockertypes.ContainerExecInspect{}` in the `NewFakeDockerClient` method or just always return `&dockertypes.Container...`?

Apologies if I'm missing something :)

**Review by mattjmcnaughton — COMMENTED** [review]
Thanks for taking a look here :)

File: pkg/kubelet/dockershim/libdocker/fake_client.go (hunk 1 above)
Comment (krzysied) [reply pkg/kubelet/dockershim/libdocker/fake_client.go]: > From looking at the code, I'm confused about if the f.ExecInspect != nil boolean will ever evaluate to True

My assumption was that `ExecInspect` is settable parameter. If someone assigned a value, it should be returned.

>  Looking in pkg/kubelet/dockershim I'm not seeing any instances where that value is actually set.

Maybe someone is vendoring this code. So hard to say if it's not used ¯\_(ツ)_/¯

I agree that assigning `ExecInspect` in `NewFakeDockerClient` would be a nice solution. I will update the code.

Comment (krzysied) [reply pkg/kubelet/dockershim/libdocker/fake_client.go]: Done.

**Comment by krzysied** [comment]
With https://github.com/kubernetes/kubernetes/pull/84858 merged there is no need for pull-kubernetes-kubemark-e2e-gce-scale test.
@wojtek-t It seems that tide is waiting for pull-kubernetes-kubemark-e2e-gce-scale to succeed. Is there a way to remove it from PR tests?

**Comment by wojtek-t** [comment]
not that I'm aware

**Comment by wojtek-t** [comment]
the only way would be to cancel the job in prow somehow

**Comment by krzysied** [comment]
I tried to remove the test but it seems I only broke the current run.
I created copy of this PR - https://github.com/kubernetes/kubernetes/pull/84918
Let's close current PR and work the copy...
Sorry for inconvenience :/

**Pull request closed without merging**